proto/structured.proto

```diff
@@ -104,3 +105,234 @@ message CreateTableResponse {
+  // The columns forming the index key must be part of the ColumnCollection.
+  optional ColumnCollection row = 3 [(gogoproto.nullable) = false];
+}
+
+// A PutTableRowResponse is the response.
+message PutTableRowResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// A ConditionalPutTableRowRequest.
+//
+// Puts row if exp_row equals an existing row in the DB. Both exp_row and row
+// must contain the columns forming the index key, and those constructed keys
+// must be equal. Returns error and the actual value of the row in the response
+// if exp_row doesn't match the full row in the DB.
+message ConditionalPutTableRowRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ColumnCollection exp_row = 3 [(gogoproto.nullable) = false];
+  optional ColumnCollection row = 4[(gogoproto.nullable) = false];
+}
+
+// A ConditionalPutTableRowResponse. The row returned is the actual row in
+// the DB that doesn't full match exp_row in the request. Only the columns
+// forming the index key match up with those in exp_row.
+message ConditionalPutTableRowResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ValueCollection row = 2[(gogoproto.nullable) = false];
+}
+
+// An IncrementTableRequest increments the value of the columns in row.
+// row can only contain the columns forming the index key and columns
+// of type integer. Incrementing by 0 is not a noop, and create a zero value.
+message IncrementTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ColumnCollection row = 2 [(gogoproto.nullable) = false];
+}
+
+// An IncrementTableResponse returns the row with the new column values.
+message IncrementTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ValueCollection row = 2 [(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowRequest deletes the entire row. key contains the columns
+// forming the index key.
+message DeleteTableRowRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ColumnCollection key = 2 [(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowResponse.
+message DeleteTableRowResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// A DeleteTableRowRangeRequest deletes the specified rows. Both key and end_key
+// must only contain the columns forming the index key.
+message DeleteTableRowRangeRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Delete *all* entries between Key (inclusive) and EndKey
+  // (exclusive). Must be > 0
```

Comment:
I think you copied this comment from `DeleteRangeRequest`, but reworded it in a way that doesn't make sense. Move this down to just above the `max_entries_to_delete` field.